lib/tests/resources/modules/reverse_valid

```diff
@@ -73,14 +74,18 @@ def action_metadata
 end
 
 def action_string
-  params = JSON.load($stdin)["params"]
-  output = { :output => params['argument'].reverse }
-  puts output.to_json
+  args = JSON.load($stdin)
+  check_output_files(args)
+  input_args = args["input"]
+  results = { :output => input_args['argument'].reverse }
+  puts results.to_json
 end
 
 def action_hash
-  hash = JSON.load($stdin)["params"]
-  hash['output'] = hash['input'].reverse
+  args = JSON.load($stdin)
+  check_output_files(args)
+  input_args = args["input"]
+  hash['output'] = input_args['the_input'].reverse
```

Comment:
on top of that the ```resutls``` schema mandates ```output``` *and* ```input``` fields in the result: https://github.com/parisiale/pxp-agent/blob/352ff9f3d2dff47b5f6b18ea05be7434ebee5d49/lib/tests/resources/modules/reverse_valid#L60-L62